bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java

```diff
@@ -349,7 +350,7 @@ public boolean inButtonMode(int idx) {
             }
         } else if (isRGBW2 && (settings.lights != null) && (idx < settings.lights.size())) {
             ShellySettingsRgbwLight light = settings.lights.get(idx);
-            btnType = light.btnType;
+            btnType = getString(light.btnType);
```

Comment:
same, avoid potentially a NPE

Reply:
Does it only affect the new devices, or is it a general bugfix?

Reply:
I have extracted it to #17180, so that it can be backported, and we can potentially avoid `NullPointerExceptions` for other dimmers or lights as well.